packages/editor/src/components/provider/index.js

```diff
@@ -61,55 +76,75 @@ class EditorProvider extends Component {
 		} );
 	}
 
-	componentDidUpdate( prevProps ) {
-		if ( this.props.settings !== prevProps.settings ) {
-			this.props.updateEditorSettings( this.props.settings );
-		}
-	}
-
 	render() {
 		const {
 			children,
+			blocks,
+			resetEditorBlocks,
+			isReady,
+			settings,
+			meta,
+			onMetaChange,
+			reusableBlocks,
+			resetEditorBlocksWithoutUndoLevel,
 		} = this.props;
 
-		const providers = [
-			// Slot / Fill provider:
-			//
-			//  - context.getSlot
-			//  - context.registerSlot
-			//  - context.unregisterSlot
-			[
-				SlotFillProvider,
-			],
-
-			// DropZone provider:
-			[
-				DropZoneProvider,
-			],
-		];
-
-		const createEditorElement = flow(
-			providers.map( ( [ Provider, props ] ) => (
-				( arg ) => createElement( Provider, props, arg )
-			) )
+		if ( ! isReady ) {
+			return null;
+		}
+
+		const editorSettings = this.getBlockEditorSettings(
+			settings, meta, onMetaChange, reusableBlocks
 		);
 
-		return createEditorElement( children );
+		return (
+			<BlockEditorProvider
+				value={ blocks }
+				onInput={ resetEditorBlocksWithoutUndoLevel }
+				onChange={ resetEditorBlocks }
+				settings={ editorSettings }
+			>
+				{ children }
+			</BlockEditorProvider>
+		);
 	}
 }
 
-export default withDispatch( ( dispatch ) => {
-	const {
-		setupEditor,
-		updateEditorSettings,
-		updatePostLock,
-	} = dispatch( 'core/editor' );
-	const { createWarningNotice } = dispatch( 'core/notices' );
-
-	return {
-		setupEditor,
-		updateEditorSettings,
-		updatePostLock,
-		createWarningNotice,
-	};
-} )( EditorProvider );
+export default compose( [
+	withSelect( ( select ) => {
+		const {
+			__unstableIsEditorReady: isEditorReady,
+			getEditorBlocks,
+			getEditedPostAttribute,
+			__experimentalGetReusableBlocks,
+		} = select( 'core/editor' );
+		return {
+			isReady: isEditorReady(),
+			blocks: getEditorBlocks(),
+			meta: getEditedPostAttribute( 'meta' ),
+			reusableBlocks: __experimentalGetReusableBlocks(),
+		};
+	} ),
+	withDispatch( ( dispatch ) => {
+		const {
+			setupEditor,
+			updatePostLock,
+			resetEditorBlocks,
+			editPost,
+		} = dispatch( 'core/editor' );
+		const { createWarningNotice } = dispatch( 'core/notices' );
+
+		return {
+			setupEditor,
+			updatePostLock,
+			createWarningNotice,
+			resetEditorBlocks,
+			resetEditorBlocksWithoutUndoLevel( blocks ) {
+				resetEditorBlocks( blocks, false );
```

Comment:
We should really avoid boolean arguments. There's little chance the next maintainer will know what the second argument here represents.